ZeeHomeScreen/Classes/ZeeHomeScreenMain.swift

```diff
@@ -47,6 +49,7 @@ public class ZeeHomeScreenMain: NSObject, ZPPluggableScreenProtocol {
     public func createScreen() -> UIViewController {
         let bundle = Bundle.init(for: type(of: self))
         let result = SectionCompositeViewController(nibName: "SectionCompositeViewController", bundle: bundle)
+        NotificationCenter.default.addObserver(result, selector: #selector(SectionCompositeViewController.screenPickerTabSelected), name: NSNotification.Name(rawValue: ZeeHomeScreenMain.kScreenPickerTabSelected), object: nil)
```

Comment:
The method `reportTabVisitedAnalyticsIfApplicable` has code for determining whether the tab was visited.

```
        guard let atomFeedUrl = atomFeedUrl else { return }
        guard let base64Url = findQueryStringParameter(url: atomFeedUrl, parameter: "url") else { return }
        guard let dataSourceUrl = decodeBase64(from: base64Url) else { return }
        guard let screenType = findQueryStringParameter(url: dataSourceUrl, parameter: "screen_type") else { return }
...
```
Please extract method and use it both for analytics and pausing / continuing the ads.